### PR TITLE
GEOMESA-1842 Move GM Test to geomesa-test module

### PR DESCRIPTION
**Pull request by aheyne**
Signed-off-by: Austin Heyne <aheyne@ccri.com>

### CODE REVIEW

File: docs/pom.xml [hunk 1]
```diff
@@ -28,54 +28,10 @@
 
     <profiles>
         <profile>
-            <id>html</id>
```
Comment (elahrvivaz) [inline docs/pom.xml]: are these docs profiles not used? @mdzimmerman ?

File: docs/pom.xml [hunk 2]
```diff
@@ -105,6 +61,54 @@
                 </plugins>
             </build>
         </profile>
+        <profile>
+            <activation>
+                <activeByDefault>false</activeByDefault>
+            </activation>
+            <id>gm-test-auto</id>
```
Comment (elahrvivaz) [inline docs/pom.xml]: i think i would move these test profiles into the test module pom

Comment (aheyne) [inline docs/pom.xml]: Gah, ignore this git failed me again...

File: pom.xml [hunk 3]
```diff
@@ -41,6 +41,7 @@
         <module>geomesa-security</module>
         <module>geomesa-spark</module>
         <module>geomesa-stream</module>
+        <module>geomesa-test</module>
```
Comment (aheyne) [inline pom.xml]: I'm using filtering from the maven assembly plugin to make things smoother.

Comment (aheyne) [inline pom.xml]: Additionally there was talk about moving these test to jars later to give us better test validation.

**Pull request merged**